File: Camera2Basic/app/src/main/java/com/example/android/camera2/basic/utils/SwipeGestureDetector.java
package com.example.android.camera2.basic;

import android.util.Log;
import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;

import androidx.navigation.Navigation;

import javax.annotation.Nullable;

public class SwipeGestureDetector extends GestureDetector.SimpleOnGestureListener {
    private static final int MIN_SWIPE_DISTANCE_X = 100;
    private static final String TAG = SwipeGestureDetector.class.getSimpleName();
    private View view;

    @Override
    public boolean onFling(
            @Nullable MotionEvent e1, @Nullable MotionEvent e2, float velocityX, float velocityY) {
        if (e1 == null || e2 == null) {
            Log.e(TAG, "e1 or e2 is null!");
            return super.onFling(e1, e2, velocityX, velocityY);
        }
        float deltaX = e1.getX() - e2.getX();
        float deltaXAbs = Math.abs(deltaX);
        if (deltaXAbs >= MIN_SWIPE_DISTANCE_X && deltaX < 0) {
            if (view.getTag().equals("right"))
                Navigation.findNavController(view).navigate(R.id.action_camera_to_video);
        }

        return true;
    }

    public SwipeGestureDetector(View view) {
        this.view = view;
    }
}

File: CameraXBasic/java/com/android/example/cameraxbasic/utils/SwipeGestureDetector.kt
/*
* Copyright 2021 NXP
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*      http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

package com.android.example.cameraxbasic.utils

import android.view.GestureDetector
import android.view.MotionEvent
import kotlin.math.abs

class SwipeGestureDetector : GestureDetector.SimpleOnGestureListener() {
    companion object {
        private const val MIN_SWIPE_DISTANCE_X = 100
    }

    var swipeCallback: SwipeCallback? = null

    override fun onFling(e1: MotionEvent?, e2: MotionEvent?, velocityX: Float, velocityY: Float): Boolean {
        if (e1 == null || e2 == null) return super.onFling(e1, e2, velocityX, velocityY)
        val deltaX = e1.x - e2.x
        val deltaXAbs = abs(deltaX)

        if (deltaXAbs >= MIN_SWIPE_DISTANCE_X) {
            if (deltaX > 0) {
                swipeCallback?.onLeftSwipe()
            } else {
                swipeCallback?.onRightSwipe()
            }
        }

        return true
    }

    interface SwipeCallback {
        fun onLeftSwipe()

        fun onRightSwipe()
    }

    fun setSwipeCallback(left: ()-> Unit = {}, right: ()-> Unit = {}) {
        swipeCallback = object : SwipeCallback {
            override fun onLeftSwipe() {
                left()
            }

            override fun onRightSwipe() {
                right()
            }
        }
    }
}
